core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -99,6 +101,28 @@ public function load(array $configs, ContainerBuilder $container): void
         ;
     }
 
+    /**
+     * Validates and sets the "contao.image.sizes" parameter.
+     */
+    private function setPredefinedImageSizes(array $config, ContainerBuilder $container): void
+    {
+        if (!isset($config['image']['sizes']) || count($config['image']['sizes']) === 0) {
+            return;
+        }
+
+        $imageSizes = [];
+
+        foreach ($config['image']['sizes'] as $name => $value) {
+            $imageSizes['_' . $name] = $value;
+        }
+
+        $services = ['contao.image.image_sizes', 'contao.image.image_factory', 'contao.image.picture_factory'];
+
+        foreach ($services as $service) {
+            $container->getDefinition($service)->addMethodCall('setPredefinedSizes', [$imageSizes]);
```

Comment:
This is actually a BC break I think because `image_factory` and `picture_factory` have an interface that doesn’t include the `setPredefinedSizes()` method. We would have to check if the services are an instanceof `ImageFactory` and `PictureFactory` and only add the method call then.

Reply:
Good catch, I have adjusted the code accordingly.

Reply:
I don't think this is the right way to go. That means I cannot replace it while still have predefined sizes :) Imho we should use `method_exists` just like Symfony does in many places too :)

Reply:
Adjusted again 😉